apps/api/src/auth/auth.service.ts

```diff
@@ -92,6 +87,51 @@ export class AuthService {
       )
   }
 
+  async issueTokenFromProvider(
+    providerDto: ProviderDto,
+  ): Promise<Observable<ApiTokenResponse | ErrorResponse>> {
+    const data = {
+      grant_type: <const>'urn:ietf:params:oauth:grant-type:token-exchange',
+      subject_token: providerDto.providerToken,
+      subject_issuer: providerDto.provider,
+      subject_token_type: 'urn:ietf:params:oauth:token-type:access_token',
+    }
+    const $tokenObs = await this.tokenEndpoint(data)
+    const keycloakResponse = await firstValueFrom($tokenObs)
+    const userInfo = await this.keycloak.grantManager.userInfo<string, KeycloakTokenParsed>(
+      keycloakResponse.accessToken as string,
+    )
+    await this.prismaService.person.upsert({
+      // Create a person with the provided keycloakId
+      create: {
+        email: userInfo.email || '',
+        firstName: userInfo.given_name || '',
+        lastName: userInfo.family_name || '',
+        picture: providerDto.picture,
+        keycloakId: userInfo.sub,
+      },
+      // Store keycloakId to the person with same email
+      update: { keycloakId: userInfo.sub },
+      where: { email: userInfo.email },
+    })
+    return $tokenObs
```

Comment:
There might be a better way to do all these with `rxjs` operators, but I am not the best there. If you have any suggestions on how to do that I am open